test/test-client-server.js

```diff
@@ -1157,7 +1157,61 @@ var tests = [
     },
     what: 'client auto-rejects unrequested, allows requested forwarded-tcpip'
   },
+  { run: function() {
+      var self = this,
+          what = this.what,
+          client,
+          server,
+          wasReply = false,
+          fastrejectSent = false,
+          r;
+
+      function sendAcceptLater(accept) {
+        // Only call accept after reject was called
+        if (fastrejectSent) {
+          accept();
+        } else {
+          process.nextTick(sendAcceptLater, accept);
```

Comment:
I had a comment somewhere in the code wondering about that at one point, but I've always leaned more towards not doing it because it could be unexpected. Even if such a timeout was configurable, it'd mean adding more overhead (adding timers) and even then you might not want the same timeout for every global request.